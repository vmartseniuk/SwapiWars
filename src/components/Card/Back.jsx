import React, { memo } from "react";
/* eslint-disable react/prop-types */
import { BackSide } from "react-flippy";
import makeStyles from "@material-ui/core/styles/makeStyles";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
/* @core */
import colors from "config/assets";
import convertToRoman from "utils/toRoman";

const useStyles = makeStyles(() => ({
  front: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    backgroundColor: "#202020",
    borderRadius: 20,
    boxShadow: `0 3px 20px -8px rgba(${colors.whiteRGB}, 0.42), 0 4px 5px 0px rgba(${colors.whiteRGB}, 0.12), 0 3px 1px -5px rgba(${colors.whiteRGB}, 0.2)`,
  },
  title: {
    fontSize: "16px",
    textAlign: "center",
    letterSpacing: "5px",
    textTransform: "uppercase",
  },
  opening: {
    padding: "15px 0",
    fontSize: "14px",
    textTransform: "uppercase",
    textAlign: "center",
  },
  button: {
    fontWeight: 300,
    fontSize: "12px",
    lineHeight: "14px",
    letterSpacing: "2px",
    textTransform: "none",
    background: "#272727",
  },
}));

const Back = (props) => {
  const { opening_crawl: opening, episode_id: episode, onClick } = props;
  const classes = useStyles();

  return (
    <BackSide className={classes.front}>
      <Typography className={classes.title}>
        {convertToRoman(episode)}
        &nbsp; episode
      </Typography>

      <Typography className={classes.opening}>{opening}</Typography>
      <Button className={classes.button} variant="outlined" size="large" onClick={onClick}>
        show more
      </Button>
    </BackSide>
  );
};
export default memo(Back);
