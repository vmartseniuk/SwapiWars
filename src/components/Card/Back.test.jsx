import { create } from "react-test-renderer";
import Back from "./Back";

describe("Card/Back", () => {
  it("renders correctly", () => {
    const props = { opening_crawl: "", episode_id: 1, onClick: () => null };
    const rendered = create(<Back {...props}>test name</Back>);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
