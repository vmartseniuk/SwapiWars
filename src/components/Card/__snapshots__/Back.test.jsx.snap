// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card/Back renders correctly 1`] = `
<div
  className="flippy-card flippy-back makeStyles-front-1"
  style={
    Object {
      "transitionDuration": "0.6s",
    }
  }
>
  <p
    className="MuiTypography-root makeStyles-title-2 MuiTypography-body1"
  >
    I
    Â  episode
  </p>
  <p
    className="MuiTypography-root makeStyles-opening-3 MuiTypography-body1"
  >
    
  </p>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-4 MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      show more
    </span>
  </button>
</div>
`;
